block-producer/src/config.rs

```diff
@@ -1,31 +1,29 @@
-use std::fmt::Display;
+use std::fmt::{self, Display, Formatter};
 
+use miden_node_store::config::StoreEndpoint;
+use miden_node_utils::config::Config;
 use serde::{Deserialize, Serialize};
 
 pub const HOST: &str = "localhost";
 // defined as: sum(ord(c)**p for (p, c) in enumerate('miden-block-producer', 1)) % 2**16
 pub const PORT: u16 = 48046;
-pub const ENV_PREFIX: &str = "MIDEN_BLOCK_PRODUCER";
 pub const CONFIG_FILENAME: &str = "miden-block-producer.toml";
 
+// Endpoint
+// ================================================================================================
+
+/// BlockProducer default endpoint.
+///
+/// Struct is specific to the block producer since it defaults to its host and port.
 #[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Serialize, Deserialize)]
-pub struct Endpoint {
-    /// The host device the server will bind to.
+pub struct BlockProducerEndpoint {
```

Comment:
We currently have 3 structs that are the same format: `RpcEndpoint`, `StoreEndpoint` and `BlockProducerEndpoint`. We encode the value they "should have" by implementing `Default` properly on them.

I think what we want to do is have a single `Endpoint` struct:

```rust
/// Represents an endpoint
pub struct Endpoint {
    pub host: String,
    pub port: u16,
}

impl Endpoint {
  // implement methods that are useful for an `Endpoint` *once* instead of 
  // copy/pasting it over to all 3 types
}

// This can also be in an `as_url()` function
impl Display for Endpoint {
  // print as `http://{host}:{port}`
}
```

Any change that we make to `Endpoint` will only need to be made once (as opposed to 3 times). For example, we might at some point use, say `https` instead of `http` (or whatever).

Then each respective config can default to the proper value

```rs
pub struct BlockProducerConfig {
    /// Block producer listen endpoint.
    pub endpoint: Endpoint,

    pub store_endpoint: Endpoint,
}

impl BlockProducerConfig {
  pub fn endpoint() -> Endpoint {
      Endpoint { /* block producer's endpoint */ }
  }
}

impl Default for BlockProducerConfig {
    fn default() -> Self {
        Self {
            endpoint: Self::endpoint(),
            store_endpoint: StoreConfig::endpoint(),
        }
    }
}
```

We also no longer need to use a `String` to represent an endpoint, as is currently used for `BlockProducerConfig::store_endpoint`.

Reply:
- Renamed `Endpoint` to `HostPort` to avoid confusion
- De-duplicated implementation, moved it to `miden-node-utils`
- Added the `as_endpoint` method to implement the default value